src/crate/qa/tests.py

```diff
@@ -186,12 +186,15 @@ def _new_cluster(self, version, num_nodes, settings=None, env=None):
             self.assertFalse(port in settings)
         s = {
             'cluster.name': gen_id(),
-            'discovery.zen.ping.unicast.hosts': self._unicast_hosts(num_nodes),
-            'discovery.zen.minimum_master_nodes': math.floor(num_nodes / 2.0 + 1),
             'gateway.recover_after_nodes': num_nodes,
             'gateway.expected_nodes': num_nodes,
             'node.max_local_storage_nodes': num_nodes,
         }
+        crate_dir = get_crate(version)
+        version_tuple = _extract_version(crate_dir)
+        if version_tuple < (4, 0, 0):
+            s['discovery.zen.ping.unicast.hosts'] = self._unicast_hosts(num_nodes)
```

Comment:
As all nodes will be bound to the loopback iface, this isn't needed for < 4.x as well afaik. Otherwise we would have to add this for >= 4.x as well under the new name `discovery.seed_hosts`.